table/table.h

```diff
@@ -40,45 +43,106 @@ class Table {
         vector<::database::Row> filter(vector<string> &columns, 
                                         vector<::database::Entry> &comparisons);
         
+        // Finds and edits rows where each entry@each column = the said comparison
+        // Changes rows to set entry@each edit_column = the said new entry
         // Input: vector<string> columns -> columns being accessed
         // Input: vector<Entry> comparisons -> list of entries to compare each row entry@column to
+        // Input: vector<string> &edit_columns -> columns being edited in filtered entries
+        // Input: vector<::database::Entry> &entries -> new entries to update entries in edit_columns
         void edit_rows(vector<string> &columns, 
                         vector<::database::Entry> &comparisons, 
                         vector<string> &edit_columns, 
                         vector<::database::Entry> &entries);
 
+        // Edits all rows to set entry@each edit_column = the said new entry
+        // Input: vector<string> &edit_columns -> columns being edited in filtered entries
+        // Input: vector<::database::Entry> &entries -> new entries to update entries in edit_columns
+        void edit_all(vector<string> &edit_columns, 
+                        vector<::database::Entry> &entries);
+
         // Inserts row into database
         // Checks if row aligns with correct types
         // Input: ::database::row &row -> row being inserted
         void insert(::database::Row &row);
         
+        // Removes rows where each entry@each column = the said comparison
+        // Input: vector<string> columns -> columns being accessed
+        // Input: vector<Entry> comparisons -> list of entries to compare each row entry@column to
+        void remove_rows(vector<string> &columns, 
+                            vector<::database::Entry> &comparisons);
+        
+        // Removes all rows in table
+        void remove_all() {
+            table.mutable_rows()->erase(table.mutable_rows()->begin(), table.mutable_rows()->begin() + table.rows_size());
+        }
+
         // Serializes table to ostream
         // Input: ostream* out -> output ostream
         bool serialize(ostream* out) {
             return table.SerializeToOstream(out);
         }
 
-        // Returns type as string given column name
+        // Returns type of column as string given column name
         // Input: string &column -> column name
         string get_type(string &column);
 
+        // Outputs Table row by row, entry by entry
         void output();
 
+        // Outputs given entry
+        // Input: ::database::Entry& entry -> entry being output
         void output_entry(::database::Entry& entry);
 
+        // Error caused by mismatch between column and entry through comparison, insertion, or updation
+        struct type_mismatch : public std::exception {
+            enum error_type {
+                comparison = 0,
```

Comment:
When defining enum values, name each value in all caps

e.g.
`COMPARISON = 0`

Using this convention will make it easier to read code and recognize something must be an enum